src/Core/Content/Mail/Message/SendMailHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Mail\Message;
+
+use League\Flysystem\FilesystemOperator;
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+use Symfony\Component\Mime\Email;
+
+#[AsMessageHandler]
+#[Package('services-settings')]
+class SendMailHandler
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly TransportInterface $transport,
+        private readonly FilesystemOperator $filesystem,
+        private readonly LoggerInterface $logger
+    ) {
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     */
+    public function __invoke(SendMailMessage $message): void
+    {
+        $mailDataPath = $message->getMailDataPath();
+
+        if (!$this->filesystem->fileExists($mailDataPath)) {
+            $this->logger->error('An error occurred while sending an email. The mail data file does not exist.');
+
+            return;
+        }
+
+        $mailData = $this->filesystem->read($mailDataPath);
+
+        $mail = unserialize($mailData);
+
+        if (!($mail instanceof Email)) {
+            $this->filesystem->delete($mailDataPath);
+
+            $this->logger->error('An error occurred while sending an email. The mail data is invalid.');
+
+            return;
+        }
+
+        $this->transport->send($mail);
+
+        try {
+            $this->filesystem->delete($mailDataPath);
+        } catch (\Throwable) {
+            // Ignore any potential errors while deleting the file, as a error would reschedule the message and send the mail again
```

Comment:
We should log it, shouldn't we? Or throw a [UnrecoverableMessageHandlingException](https://symfony.com/doc/current/messenger.html#avoiding-retrying)

Reply:
"If you throw UnrecoverableMessageHandlingException, the message will not be retried."

Reply:
Yep, sorry, I clicked on the link too late. I read it and am now adjusting it.

Reply:
ah commented something similiar on the other log statements

But is there any reason not using the UnrecoverableMessageHandlingException? as that would make debugging a lot easier as the failed messages should be persistent in the failed transport and could be inspected manually 🤔 

at least for everything before message sending, when we send the mail I guess it is correct to let the handler exit successfully even if we can not clean up

Reply:
I changed it.